### PR TITLE
Add Python things

### PR DESCRIPTION
**Pull request by Simown**
Added python interface and loading/scripting functionality.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,12 @@
 #              will be harmless on other machines.
 ######################################################################
 
+LUA_LINK=
```
Comment (jpirie) [inline Makefile]: I guess we need something like  -llua -ldl here to include the lua libraries.

File: lua-interface.cpp [hunk 2]
```diff
@@ -112,10 +108,12 @@ int LuaInterface::sendLuaPrivateMessage(lua_State *luaState) {
 
 LuaInterface::LuaInterface(string thisSaveFile) {
   saveFile = thisSaveFile;
+  firstPluginLoad = true;
```
Comment (jpirie) [inline lua-interface.cpp]: Do we have this variable defined anywhere now? Did I miss something?

File: python-interface.cpp [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+      PyErr_Print();
+      Py_RETURN_FALSE;
+  }
+
+  // TODO: Should be able to parse it as a char or actually an int/bool and skip this conversion
+  bool isPrivateMessage = (*isPrivate != '0');
+
+  string msg = ircbot->getChannelSendString();
+  msg  += message;
+  msg += "\r\n";
```
Comment (jpirie) [inline python-interface.cpp:142]: I guess at some point we should deal with carriage-return line-feed somewhere nice. Debugging the problem with the PING/PONG messages over the wire with these everywhere was pretty sad.

File: Makefile (hunk 1 above)
Comment (Simown) [inline Makefile]: Yes, you are right. I need a few more lines on my local copy but not on Kiwi. We do need them loads, yes. I'll add them in.

File: python-interface.cpp (hunk 3 above)
Comment (Simown) [inline python-interface.cpp:142]: Yeah, very true. Python scripting implementation copied some of the bad habits just to get it working. The whole interface(s) need a tidy up at some point.

File: python/plugins/roll.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        rollMatch = rollPattern.match(str(message))
+        if(rollMatch):
+            output = "Rolling " + str(rollMatch.group(1)) + ". "
+            multiplier = (rollMatch.group(2))
+            dice = int(rollMatch.group(3))
+            op = rollMatch.group(4)
+            mod = rollMatch.group(5)
+            if(not multiplier):
+                multiplier = 1
+            if(dice < 1):
```
Comment (jpirie) [inline python/plugins/roll.py]: I guess you could add in the condition to stop kiwi looping for millions of dice with !roll 1000000d6 or something if you wanted. Whatever you want really, the implementation is just for us anyways. :-)

Comment (Simown) [inline python/plugins/roll.py]: Updated roll plugin coming soon! Needs more testing locally first. New one won't accept more than 1000 dice as kiwi can't hold that many (obviously) plus some nifty other features.

**Pull request merged**